TESTING.md

```diff
@@ -0,0 +1,274 @@
+  "Success!" yet...
+- Run internal `chai` + `chai-jquery` matchers (`expect($el).to.contain('Success!')`) until an error
+  is no longer thrown. `.should` will rerun the previous command _and_ the assertion until the
+  condition is met or a timeout occurs. This is the secret sauce to the stability of Cypress tests
+  compared to WebDriver. `.should` can also receive a function that will be run until no error is
+  thrown. If an error is never thrown, it will only be run once.
+
+Cypress is very easy to get started with, but there isn't enough documentation about how to scale a
+complicated suite of Cypress tests.
+
+### Component helpers
+
+Component helpers should be written to help with automation. These component helpers are useful for
+a more readable implementation as well as more maintainable test code. Component helpers can also be
+exported by this repository for use in downstream testing (Cypress tests in applications that use
+Canvas Kit). Helpers contain implementation details about how components are composed internally.
+The goal is that if the Component implementation changes, the component helpers should be the only
+code that needs to change to keep tests passing.
+
+Component helpers come in 3 flavors:
+
+- **Starter**: This helper types starts a Cypress chain. The return type is always
+  `Cypress.Chainable<JQuery>` and usually contains a `cy.get` and the selector usually contains
+  something that is unique to that component type. For example, the Modal helper uses
+  `[role=dialog]` with an option to disambiguate with a `data-testid`.
+- **Transform**: Imagine a web application is a collection of boxes that determine how components of
+  an application are laid out. A transform helper takes in a larger box and returns a smaller box,
+  further reducing the scope. Ideally these helpers contain no Cypress commands (which are
+  asynchronous), only jQuery calls (which are synchronous). Synchronous helper functions can be run
+  more than once at the discretion of the Cypress runtime. A good use-case for this type is to
+  replace CSS selectors
+- **Action**: Cypress comes with many baked in action commands such as `click` and `type`. A custom
+  action is useful to give a higher level semantic meaning to a series of low-level commands. For
+  example, a Select component might compose a Popup and a virtualized menu. This action may include
+  opening the select, finding the popup element, seeing if the desired item is currently visible,
+  scrolling the virtualized menu until the desired item is visible and clicking on the item. This
+  code would be repeated for every instance and has implementation knowledge of the component. A
+  high-level semantic action might be called "select" that wraps up this implementation detail to
+  free up automation and development engineers to think about the test case and not about
+  implementation details.
+
+### How to write a functional test
```

Comment:
This sounds like it should go somewhere else in the document, some of the content below is repeated, is this intentional or still part of the draft process?

Reply:
I think this should go somewhere else... This section isn't related to the 3 levels of testing and what should go where... I think we could remove it from this document and it should be a different one

Reply:
The only content I think is relevant to this document is that both visual regression and functional tests use stories as a unit to test against.

Reply:
I added info about how Storybook stories relate to functional and visual tests in the intro to the levels. I think I can remove this last section and save it for another document.

Functional testing is probably the broadest topic with the most nuances. It is also generally less understood than unit testing.